lib/sort-order-selector/index.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { Fragment, FunctionComponent, useEffect, useRef } from 'react';
+import { connect } from 'react-redux';
+import { toggleSortOrder, setSortType } from '../state/settings/actions';
+import { recordEvent } from '../state/analytics/middleware';
+
+import * as S from '../state';
+import * as T from '../types';
+
+type StateProps = {
+  shouldDisplay: boolean;
+  sortReversed: boolean;
+  sortType: T.SortType;
+};
+
+type DispatchProps = {
+  setSortType: (sortType: T.SortType) => any;
+  toggleSortOrder: () => any;
+};
+
+type Props = StateProps & DispatchProps;
+
+export const SortOrderSelector: FunctionComponent<Props> = ({
+  shouldDisplay,
+  setSortType,
+  sortReversed,
+  sortType,
+  toggleSortOrder,
+}) => {
+  const changeSort = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedOption = event.currentTarget.selectedOptions[0];
+    const selectedSortType = selectedOption.dataset.order;
+    const selectedSortReversed = selectedOption.dataset.reversed;
+
+    if (selectedSortType !== sortType) {
+      setSortType(selectedSortType as T.SortType);
+    }
+    if ((selectedSortReversed === 'false') === sortReversed) {
+      toggleSortOrder();
+    }
+    recordEvent('list_sortbar_mode_changed', {
+      description: selectedOption.text,
+    });
+  };
+
+  const sortTypes = [
+    {
+      label: 'Name: A-Z',
+      id: 'alphabetical',
+      order: 'alphabetical',
+      reversed: false,
+    },
+    {
+      label: 'Name: Z-A',
+      id: 'alphabetical-reversed',
+      order: 'alphabetical',
+      reversed: true,
+    },
+    {
+      label: 'Created: Newest',
+      id: 'creationDate',
+      order: 'creationDate',
+      reversed: false,
+    },
+    {
+      label: 'Created: Oldest',
+      id: 'creationDate-reversed',
+      order: 'creationDate',
+      reversed: true,
+    },
+    {
+      label: 'Modified: Newest',
+      id: 'modificationDate',
+      order: 'modificationDate',
+      reversed: false,
+    },
+    {
+      label: 'Modified: Oldest',
+      id: 'modificationDate-reversed',
+      order: 'modificationDate',
+      reversed: true,
+    },
+  ];
```

Comment:
if we want to simplify the flow of data and types through the select component then we can do another thing and that is directly index into this list. consider if this `sortTypes` array were outside of the component and in the module scope and we had a function to find the current setting. our `HTMLSelectElement` could refer to an index into the array as the choice, and although that's less meaningful in itself, it does preserve the types the eliminates the processing and parsing needed to re-connect the data.

```ts
type SortOption = {
	label: string,
	order: T.SortOrder,
	isReversed: boolean
}

const sortTypes: SortOption[] = [{
	label: 'Name: A-Z',
  order: 'alphabetical',
  isReversed: false
}, {
	…
}];

const onChange = ({currentTarget: {selectedIndex}}) => {
	if (-1 === selectedIndex) {
		return;
	}

	const sort = sortTypes[selectedIndex];
	if (sort.sortOrder !== sortOrder) {
		…
	}

	if (sort.isReversed !== sortReversed) {
		…
	}
}

const sortTypesIndex = (sortOrder, isReversed) => sortTypes.findIndex(type => type.order === sortOrder && type.isReversed === isReversed);

<select id="sort-selection" value={sortTypesIndex(sortOrder, reversed)} onChange={…}>
{sortTypes.map((type, index) => (
	<option key={type.label} value={index}>{type.label}</option>
))}
</select>
```

Reply:
it's _rare_ to use an index as the React key but here since the `sortTypes` array is static the indices will never change and it's safe to use that value

Reply:
That works like a charm! https://github.com/Automattic/simplenote-electron/pull/2542/commits/a7d6f62510933d1486c3d79530eb3dfbd2c171f8